frontend/src/pages/modelServing/customServingRuntimes/CustomServingRuntimeAddTemplate.tsx

```diff
@@ -33,13 +36,28 @@ const CustomServingRuntimeAddTemplate: React.FC<CustomServingRuntimeAddTemplateP
 }) => {
   const { dashboardNamespace } = useDashboardNamespace();
   const { refreshData } = React.useContext(CustomServingRuntimeContext);
-  const { state } = useLocation();
+  const { state }: { state?: { template: TemplateKind } } = useLocation();
+
+  const copiedServingRuntimeString = state
+    ? YAML.stringify({
+        ...state.template.objects[0],
+        metadata: {
+          ...state.template.objects[0].metadata,
+          name: `${getServingRuntimeNameFromTemplate(state.template)}-copy`,
+          annotations: {
+            'openshift.io/display-name': `Copy of ${getServingRuntimeDisplayNameFromTemplate(
+              state.template,
+            )}`,
+            'openshift.io/description':
+              state.template.objects[0].metadata.annotations?.['openshift.io/description'],
+          },
+        },
+      })
+    : '';
 
   const stringifiedTemplate = existingTemplate
     ? YAML.stringify(existingTemplate.objects[0])
-    : state
-    ? YAML.stringify(state.template)
-    : '';
+    : copiedServingRuntimeString;
```

Comment:
With this change, the user who is duplicating a template cannot immediately perform a save even though it is a valid resource since the `metadata.name` was modified with a suffix `-copy`.

Reply:
Good catch!